### PR TITLE
[Storage] Support share soft delete 

### PR DESCRIPTION
**Pull request by blueww**
## Description

design： Azure/azure-powershell-cmdlet-review-pr#559
the change has been in storage preview before with PR  https://github.com/Azure/azure-powershell/pull/11779

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Storage/Storage.Management/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,13 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Support enable/disable/get share soft delete properties on file Service of a Storage account
```
Comment (VeryEarly) [inline src/Storage/Storage.Management/ChangeLog.md:21]:
```suggestion
* Supported enable/disable/get share soft delete properties on file Service of a Storage account
```

**Review by VeryEarly — CHANGES_REQUESTED** [review]

**Comment by VeryEarly** [comment]
please fix test failures

**Comment by blueww** [comment]
Tried to fix the CI failures.
And add a static exception for RetentionDays , the name has been discussed in
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/726

**Comment by blueww** [comment]
@VeryEarly 
Would you please help to look at the test timeout on windows? (all other CI passed)

The hang happens seems in SQL test, which seems not related with my change.
Would you please help to check?

##### Test result:
 https://dev.azure.com/azure-sdk/public/_build/results?buildId=553654&view=logs&jobId=b17395f6-68a3-5682-0476-d3f6f1043109&j=b17395f6-68a3-5682-0476-d3f6f1043109&t=67f633ab-4360-5839-af42-f446e0c847f0

##### Test log: 
> 2020-09-27T02:37:43.3882233Z   [xUnit.net 00:08:37.15]     Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ServerDisasterRecoveryConfigurationTests.TestServerDisasterRecoveryConfiguration [SKIP]
> 2020-09-27T02:37:43.4295338Z     ! Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ServerDisasterRecoveryConfigurationTests.TestServerDisasterRecoveryConfiguration [1ms]
> 2020-09-27T02:39:13.0826785Z   [xUnit.net 00:10:06.85]     Microsoft.Azure.Commands.Sql.Test.ScenarioTests.VulnerabilityAssessmentTests.VulnerabilityAssessmentWithSettingsNotDefinedTest [SKIP]
> 2020-09-27T02:39:13.3020838Z     ! Microsoft.Azure.Commands.Sql.Test.ScenarioTests.VulnerabilityAssessmentTests.VulnerabilityAssessmentWithSettingsNotDefinedTest [1ms]
> 2020-09-27T05:09:42.1595873Z ##[error]The operation was canceled.
> 2020-09-27T05:09:42.1818638Z ##[section]Finishing: Test

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**